components/scream/src/share/io/scream_output_manager.cpp

```diff
@@ -0,0 +1,123 @@
+  //     OUT_N: INT
+  //     OUT_OPTION: STRING
+  // ----
+  // where OUT_OPTION is the units of the output frequency (ex. Steps, Months, etc.),
+  // and OUT_N>0 is the actual frequency. E.g., OUT_N=2 meanse every other ${OUT_OPTION}
+
+  // The output history should have the same restart frequency as the model restart output.
+  int restart_freq = 0;
+  std::string restart_freq_units = "None";
+  if (m_params.isSublist("Restart Control")) {
+    // Get restart parameters, and create a param list for the model restart output
+    auto& out_params = m_params.sublist("Restart Control");
+    make_restart_param_list(out_params);
+    new_output(out_params,true);
+
+    // Gather restart frequency info for restart history files
+    auto& freq_params = out_params.sublist("FREQUENCY");
+    restart_freq = freq_params.get<Int>("OUT_N");
+    restart_freq_units = freq_params.get<std::string>("OUT_OPTION");
+  }
+
+  // Construct and store an output stream instance for each output request.
+  // Typical output is controlled by parameter list which is stored in individual YAML files.
+  // A list of those files is passed to the output manager via the parameter list using the
+  // key "Output YAML Files".  See srd/share/io/tests for examples.
+  std::vector<std::string> empty;
+  auto& list_of_files = m_params.get("Output YAML Files",empty);
+  for (auto& it : list_of_files) {
+    ekat::ParameterList out_params(it);
+    parse_yaml_file(it,out_params);
+    out_params.set("CHECKPOINT FREQUENCY",restart_freq);
+    out_params.set("OUT_OPTION",restart_freq_units);
+    new_output(out_params,false);
+  }
+}
+/*===============================================================================================*/
+void OutputManager::run(util::TimeStamp& current_ts)
+{
+  for (auto& it : m_output_streams) {
+    it->run(current_ts);
+  }
+}
+/*===============================================================================================*/
+void OutputManager::finalize()
+{
+  for (auto& it : m_output_streams) {
+    it->finalize();
+  }
+  m_output_streams.clear();
+  m_io_comm = ekat::Comm();
+  m_params = ekat::ParameterList("");
+  m_field_mgr = nullptr;
+  m_runtype_restart = false;
+}
+/*===============================================================================================*/
+void OutputManager::
+new_output(const ekat::ParameterList& params, const bool model_restart_output)
+{
+  auto output = std::make_shared<output_type>(m_io_comm,params,m_field_mgr,
+                                              m_runtype_restart, model_restart_output);
+  m_output_streams.push_back(output);
+}
+/*===============================================================================================*/
+void OutputManager::make_restart_param_list(ekat::ParameterList& params)
+{
+  // Given the unique nature of restart files, this routine sets up the specific requirements.
+
+  //TODO change this to some sort of case_name, TODO: control so suffix is .r.nc
```

Comment:
Probably a bigger TODO which is on the list of SCORPIO tasks is to create a generic case name for all runs that will be used for the output file names.  For example in EAM I think the pattern for all outputs is to have the name MACHINE.RESOLUTION.COMPSET.PROCESS.DATE.nc (or something along those lines)

Reply:
Let's add it to the TODO list then. It should be item 817191, right? =)

Jokes aside, a good v1.1 issue.